sqlglot/expressions.py

```diff
@@ -3321,6 +3346,12 @@ def convert(value):
             keys=[convert(k) for k in value.keys()],
             values=[convert(v) for v in value.values()],
         )
+    if isinstance(value, datetime.datetime):
+        datetime_literal = Literal.string(value.strftime("%Y-%m-%d %H:%M:%S"))
```

Comment:
i don't believe these are correct. you're hard-coding hive date formats here. it might be more generic to do

Cast(this=literal, to=DataType.build("timestamp"))

Also should you handle timezones? in that case maybe you want utc time or using iso format

this holds the same for date

i also have a method called TimeStringToTime and DateStringToDate, the issue with those functions is that i haven't implemented them for every dialect yet, although we could make those default to CAST(...)

this may be the way to go